docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+
+It might seem like returning only a single `React` component is a bit limiting. However, you can pass in child components to whatever `React` component you render. Will get more into what that actually means in the next section, but in laymens terms it means you can `render` however many components you want:
+
+```
+    var React = require('react');
+    var MyComplicatedComponent = React.createClass({
+      var someArrayOfComponents = [<ComponentA key={0} />, <ComponentB key={1} />];
+      render: function () {
+        return (
+          <div>
+            <MyComponent className='custom-class-name' />
+            <h1>Let's render something {thisComponentIsAwesome ? 'Awesome' : 'Lame'}</h1>
+            { someArrayOfComponents }
+          </div>
+        );
+      }
+    });
+```
+
+**NOTES**
+1. We are wrapping `someArrayOfComponents` and our ternary expression in `{}`. That's because, in order to use Javascript variables and expressions in our JSX we need to wrap them in `{}` and we need to use ternary expressions if we want to write if else statements in our JSX. You can learn more about if-else in JSX [here](http://facebook.github.io/react/tips/if-else-in-JSX.html).
+2. We are passing in `key` properties to our components in our array. Whenever we render an array of components we need to pass in a unique key to each one, which `React` will use to determine which components in the array have been modified and which have been deleted/added when the array updates. You can't access the `key` property from within the component itself.
+2. There are several restrictions regarding what Javascript we can write in our JSX and you can read more about that [here](http://facebook.github.io/react/docs/jsx-in-depth.html#javascript-expressions).
+3. We are passing in a property `className`. This, is because `class` is a reserved word in Javascript. See [here](http://facebook.github.io/react/docs/jsx-in-depth.html#html-tags-vs.-react-components) for some other differences between JSX and html
+4. See [here](http://facebook.github.io/react/docs/jsx-gotchas.html) for all other JSX gotchas.
+
+***
+
+#### <a id="React-Components-Accessing-the-DOM-from-React"></a>3.2) Accessing the DOM from React [#](#React-Components-Accessing-the-DOM-from-React)
+`React` provides the `getDOMNode` method for accessing the actual DOM node for a particular component. This can be valuable for interacting with other libraries such as `jQuery` or reading from inputs. Normally, to use `getDOMNode` you will need to pass in a `ref` property to the component. The `ref` property allows the owning component to access the component via `this.refs.REF`. **NOTE** you cannot access the `ref` property from within the ownee component itself.
+
+```
+    var InputForm = React.createClass({
+      handleSubmit: function (e) {
+        e.preventDefault(),
+        this.addTodo(this.refs.inputField.value);
```

Comment:
It might be worthwhile to add in the addTodo function, but it might lead to a lot of additional code and complicating a clean example.